src/components/SplitButton/SplitButton.tsx

```diff
@@ -0,0 +1,222 @@
+
+interface TriggerProps extends HTMLAttributes<HTMLButtonElement> {
+  type?: ButtonType;
+  disabled?: boolean;
+}
+
+const SplitButtonTrigger = styled.div<{ $disabled: boolean; $type: ButtonType }>`
+  display: inline-flex;
+  align-items: center;
+  overflow: hidden;
+  ${({ theme, $disabled, $type }) => `
+    border-radius: ${theme.click.button.radii.all};
+    border: 1px solid ${theme.click.button.split[$type].stroke.default};
+    ${
+      $disabled
+        ? `border-color: ${theme.click.button.split[$type].stroke.disabled};`
+        : `
+          &:hover {
+            border-color: ${theme.click.button.split[$type].stroke.hover};
+          }
+          &:focus-within {
+            border-color: ${theme.click.button.split[$type].stroke.active};
+          }
+        `
+    }
+  `}
+`;
+
+const PrimaryButton = styled.button<{ $type: ButtonType }>`
+  display: flex;
+  align-items: center;
+  border: none;
+  align-self: stretch;
+  ${({ theme, $type }) => `
+    padding: ${theme.click.button.basic.space.y} ${theme.click.button.basic.space.x};
+    gap: ${theme.click.button.basic.space.gap};
+    background: ${theme.click.button.split[$type].background.main.default};
+    font: ${theme.click.button.basic.typography.label.default};
+    color: ${theme.click.button.split[$type].text.default};
+    &:hover {
+      background: ${theme.click.button.split[$type].background.main.hover};
+      font: ${theme.click.button.basic.typography.label.hover};
+      color: ${theme.click.button.split[$type].text.hover};
+    }
+    &:focus {
+      background: ${theme.click.button.split[$type].background.main.active};
+      font: ${theme.click.button.basic.typography.label.active};
+      color: ${theme.click.button.split[$type].text.active};
+      outline: none;
+    }
+    &:disabled {
+      background: ${theme.click.button.split[$type].background.main.disabled};
+      font: ${theme.click.button.basic.typography.label.disabled};
+      color: ${theme.click.button.split[$type].text.disabled};
+    }
+  `}
+`;
+
+const SecondaryButton = styled(DropdownMenu.Trigger)<{ $type: ButtonType }>`
+  display: flex;
+  align-items: center;
+  border: none;
+  align-self: stretch;
+  ${({ theme, $type }) => `
+    padding: ${theme.click.button.split.icon.space.y} ${theme.click.button.split.icon.space.x};
+    gap: ${theme.click.button.basic.space.gap};
+    background: ${theme.click.button.split[$type].background.action.default};
+    color: ${theme.click.button.split[$type].text.default};
+    &:hover {
+      background: ${theme.click.button.split[$type].background.action.hover};
+      color: ${theme.click.button.split[$type].text.hover};
+    }
+    &:focus {
+      background: ${theme.click.button.split[$type].background.action.active};
+      color: ${theme.click.button.split[$type].text.active};
+      outline: none;
+    }
+    &[data-disabled] {
+      background: ${theme.click.button.split[$type].background.action.disabled};
```

Comment:
what about for the primary button be our Button and the secondary button be the IconButton?

if we need to style them further we can do 
```js
const SplitButton= styled(Button)``
```

Reply:
@serdec The styling is different for the normal button and the split buttons
We are not reusing it
Only the gap is reused and thats why I didn't use the Button component and IconButton

Reply:
even though the styling is different, we should share the same functionalities of our own Button. It would be weird if it had different apis or behavior